snapshots/devmapper/snapshotter.go

```diff
@@ -350,6 +369,25 @@ func (s *Snapshotter) Close() error {
 }
 
 func (s *Snapshotter) createSnapshot(ctx context.Context, kind snapshots.Kind, key, parent string, opts ...snapshots.Opt) ([]mount.Mount, error) {
+	var fileSystemType fsType
+
+	// For snapshots with no parents, we use file system type as configured in config.
+	// For snapshots with parents, we inherit the file system type. We use the same
+	// file system type derived here for building mount points later.
+	fsLabel := make(map[string]string)
+	if len(parent) == 0 {
+		fileSystemType = s.config.FileSystemType
+	} else {
+		_, snapInfo, _, err := storage.GetInfo(ctx, parent)
+		if err != nil {
+			log.G(ctx).Errorf("failed to read snapshotInfo for %s", parent)
+			return nil, err
+		}
+		fileSystemType = fsType(snapInfo.Labels[devmapperSnapshotFsType])
```

Comment:
The existing data is `ext4` type and doesn't have label `devmapperSnapshotFsType`. The `fileSystemType` will be empty. We can consider about migration. When setup, the devmapper plugin adds the label for existing data and marks schema version, like  https://github.com/containerd/containerd/blob/main/metadata/migrations.go. WDYT?

Reply:
We inherit the file system label from parent. One simple solution that comes to my mind is
a. If we see parent snapshot dont have fsType lable
    i.  read fsType by reading file system magic number,
    ii. Set label in parent based on what we find in i.
    iii. Use same fsType for the child.
Eventually we will end up having labels in all snapshots ( old and new versions ). Sounds correct?

Reply:
@fuweid Added one change to use ext4 if parent does not have any label. This should allow new snapshot still inherit the file system type and also have ext4 label for fsType. Current assumption is committed snapshot with no fsType label means it was created with older version of containerd before this label support was added and has to be ext4.

Reply:
The change about label looks good to me. Thanks!